services/discourse_translator/amazon.rb

```diff
@@ -93,6 +93,10 @@ def self.access_token_key
     end
 
     def self.detect(post)
+      text = post.cooked.truncate(MAXLENGTH, omission: nil)
+
+      return if text.blank?
```

Comment:
I think you can use `text` in line 103

Reply:
Ah. Yes. That was the whole point of the variable ... 😅 Good catch!